src/PKSim.Core/Snapshots/Mappers/ProjectMapper.cs

```diff
@@ -231,33 +233,40 @@ private Task updateProjectClassifications(SnapshotProject snapshot, SnapshotCont
 
       private void addSimulationToProject(ModelProject project, Model.Simulation simulation)
       {
-         addClassifiableToProject<ClassifiableSimulation, Model.Simulation>(project, simulation, project.AddBuildingBlock);
+         addClassifiableToProject<ClassifiableSimulation, Model.Simulation>(project, simulation, project.AddBuildingBlock, project.All<Model.Simulation>());
       }
 
       private void addObservedDataToProject(ModelProject project, ModelDataRepository repository)
       {
-         addClassifiableToProject<ClassifiableObservedData, ModelDataRepository>(project, repository, project.AddObservedData);
+         addClassifiableToProject<ClassifiableObservedData, ModelDataRepository>(project, repository, project.AddObservedData, project.AllObservedData);
       }
 
       private void addComparisonToProject(ModelProject project, ISimulationComparison simulationComparison)
       {
-         addClassifiableToProject<ClassifiableComparison, ISimulationComparison>(project, simulationComparison, project.AddSimulationComparison);
+         addClassifiableToProject<ClassifiableComparison, ISimulationComparison>(project, simulationComparison, project.AddSimulationComparison, project.AllSimulationComparisons);
       }
 
       private void addParameterIdentificationToProject(ModelProject project, ModelParameterIdentification parameterIdentification)
       {
          addClassifiableToProject<ClassifiableParameterIdentification, ModelParameterIdentification>(project, parameterIdentification,
-            project.AddParameterIdentification);
+            project.AddParameterIdentification, project.AllParameterIdentifications);
       }
 
       private void addQualificationPlanToProject(ModelProject project, Model.QualificationPlan qualificationPlan)
       {
-         addClassifiableToProject<ClassifiableQualificationPlan, Model.QualificationPlan>(project, qualificationPlan, project.AddQualificationPlan);
+         addClassifiableToProject<ClassifiableQualificationPlan, Model.QualificationPlan>(project, qualificationPlan, project.AddQualificationPlan, project.AllQualificationPlans);
       }
 
       private void addClassifiableToProject<TClassifiableWrapper, TSubject>(ModelProject project, TSubject subject,
-         Action<TSubject> addToProjectAction) where TClassifiableWrapper : Classifiable<TSubject>, new() where TSubject : IWithId, IWithName
+         Action<TSubject> addToProjectAction, IEnumerable<TSubject> existingInProject) where TClassifiableWrapper : Classifiable<TSubject>, new() where TSubject : class, IWithId, IWithName
       {
+         var existing = existingInProject.FindByName(subject.Name);
+         if (existing != null)
+         {
+            _logger.AddError(PKSimConstants.Error.SnapshotDuplicateEntryByName(subject.Name, _executionContext.TypeFor(subject)));
+            return;
```

Comment:
skipping if duplicate. Doing it for the calssifiable as well as the building blocks